### ISSUE TITLE
Convert testcases/views::edit to class-based view

### ISSUE THREAD

**Issue by asankov** [kiwitcms/Kiwi#812]
After doing that, the `NewCaseView` and the `EditCaseView` will look very similar and may be merged together. 
Also, `EditCaseForm` can be removed. Related to #708

**Comment by atodorov** [comment]
@RMadjev once you are done with your current task take a look at this one. It is similar to refactoring you have been doing towards class-based generic views and will also unblock some other items most likely.

**Comment by atodorov** [comment]
Both views have now been converted to class based and they are built on top of Django's generic views. Form leftovers will be dealt with in #708. Additional refactoring (b/c the 2 views are very similar) is out of scope for now. Closing this one.
